### PR TITLE
[#12048] Data migration for student entities

### PR DESCRIPTION
**Pull request by marquestye**
Part of #12048 

**Outline of Solution**

- Added method to seed students in `SeedDb.java`
- Added migration and verification scripts
    - `User` entities are automatically created by Hibernate along with `Student` creation
- Added `getReferenceBySimpleNaturalId` in `HibernateUtil` to fetch `Accounts`
    - speeds up (previous iteration of) migration script from 30mins to 21mins for 10k student entities
- Added student count verification to `VerifyCourseEntityCounts`

| Entity | Test counts | Migration Time | Verification Time |
|--------|-------------|-------------------|-------------------|
| Student | 100 students * 100 courses| 25m 32s | 6m 1s |

### CODE REVIEW

File: src/client/java/teammates/client/scripts/sql/DataMigrationForStudentSql.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
+package teammates.client.scripts.sql;
+
+import com.googlecode.objectify.cmd.Query;
+
+// import jakarta.persistence.criteria.CriteriaBuilder;
```
Comment (mingyuanc) [inline src/client/java/teammates/client/scripts/sql/DataMigrationForStudentSql.java:5]: Any reason why you leave them as comments?

Comment (marquestye) [reply src/client/java/teammates/client/scripts/sql/DataMigrationForStudentSql.java:5]: For convenience, to be uncommented along with private `getTeam` function below after the migration script for `Team` entities is done.

**Review by mingyuanc — APPROVED** [review]
LGTM

**Comment by NicolasCwy** [comment]
Merged in #13036

**Pull request closed without merging**